pallets/asset-handler/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+			+ Inspect<<Self as frame_system::Config>::AccountId>
+			+ Transfer<<Self as frame_system::Config>::AccountId>;
+		/// Native Currency Identifier
+		type NativeCurrencyId: Get<u128>;
+	}
+
+	// Pallets use events to inform users when important changes are made.
+	// https://docs.substrate.io/main-docs/build/events-errors/
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// New Council Member Added [new_pending_member]
+		NewPendingMemberAdded(T::AccountId),
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Storage Overflow
+		StorageOverflow,
+		/// Not a Valid Sender
+		BadOrigin,
+		/// Cannot Mint Native Asset
+		CannotMintNativeAsset,
+		/// Cannot Burn Native Asset
+		CannotBurnNativeAsset,
+	}
+
+	impl<T: Config> Inspect<T::AccountId> for Pallet<T> {
+		type AssetId = u128;
+		type Balance = u128;
+
+		fn total_issuance(asset: Self::AssetId) -> Self::Balance {
+			// when asset is not polkadex
+			return if asset != T::NativeCurrencyId::get() {
+				T::MultiCurrency::total_issuance(asset.saturated_into()).saturated_into()
+			} else {
+				T::Currency::total_issuance().saturated_into()
+			}
+		}
+
+		fn minimum_balance(asset: Self::AssetId) -> Self::Balance {
+			return if asset != T::NativeCurrencyId::get() {
+				T::MultiCurrency::minimum_balance(asset.saturated_into()).saturated_into()
+			} else {
+				T::Currency::minimum_balance().saturated_into()
+			}
+		}
+
+		fn balance(asset: Self::AssetId, who: &T::AccountId) -> Self::Balance {
+			return if asset != T::NativeCurrencyId::get() {
+				T::MultiCurrency::balance(asset.saturated_into(), who).saturated_into()
+			} else {
+				T::Currency::total_balance(who).saturated_into()
+			}
+		}
+
+		fn reducible_balance(
+			asset: Self::AssetId,
+			who: &T::AccountId,
+			keep_alive: bool,
+		) -> Self::Balance {
+			return if asset != T::NativeCurrencyId::get() {
+				T::MultiCurrency::reducible_balance(asset.saturated_into(), who, keep_alive)
+					.saturated_into()
+			} else {
+				T::Currency::free_balance(who).saturated_into()
+			}
+		}
+
+		fn can_deposit(
+			asset: Self::AssetId,
+			who: &T::AccountId,
+			amount: Self::Balance,
+			mint: bool,
+		) -> DepositConsequence {
+			return if asset != T::NativeCurrencyId::get() {
+				T::MultiCurrency::can_deposit(asset, who, amount.saturated_into(), mint)
+			} else {
+				// balance of native asset can always be increased
+				DepositConsequence::Success
+			}
+		}
+
+		fn can_withdraw(
+			asset: Self::AssetId,
+			who: &T::AccountId,
+			amount: Self::Balance,
+		) -> WithdrawConsequence<Self::Balance> {
+			return if asset != T::NativeCurrencyId::get() {
+				let consequences = T::MultiCurrency::can_withdraw(
+					asset.saturated_into(),
+					who,
+					amount.saturated_into(),
+				);
+				return consequences.into()
+			} else {
+				todo!()
```

Comment:
Is it suppose to be like this for now?

Reply:
We are not using this function anywhere.

Reply:
Shouldn't it be removed than at all? but anyways I assume we will have new review when will merge it to dev so resolving for now.